packages/cosmos/src/cosmos_builder.rs

```diff
@@ -0,0 +1,258 @@
+        self.referer_header.as_deref()
+    }
+
+    /// See [Self::referer_header]
+    pub fn set_referer_header(&mut self, referer_header: Option<String>) {
+        self.referer_header = referer_header;
+    }
+
+    /// Set the number of bb8 connections
+    pub fn connection_count(&self) -> Option<u32> {
+        self.connection_count
+    }
+
+    /// See [Self::connection_count]
+    pub fn set_connection_count(&mut self, connection_count: Option<u32>) {
+        self.connection_count = connection_count;
+    }
+
+    /// Sets the duration to wait for a connection.
+    ///
+    /// Defaults to 5 seconds
+    ///
+    /// See [bb8::Builder::connection_timeout]
+    pub fn connection_timeout(&self) -> Duration {
+        self.connection_timeout
+            .unwrap_or_else(|| Duration::from_secs(5))
+    }
+
+    /// See [Self::connection_timeout]
+    pub fn set_connection_timeout(&mut self, connection_timeout: Option<Duration>) {
+        self.connection_timeout = connection_timeout;
+    }
+
+    /// See [bb8::Builder::retry_connection]
+    pub fn retry_connection(&self) -> Option<bool> {
+        self.retry_connection
+    }
+
+    /// See [Self::retry_connection]
+    pub fn set_retry_connection(&mut self, retry_connection: Option<bool>) {
+        self.retry_connection = retry_connection;
+    }
+
+    /// Sets the number of seconds before an idle connection is reaped
+    ///
+    /// Defaults to 20 seconds
+    pub fn idle_timeout_seconds(&self) -> u32 {
+        self.idle_timeout_seconds.unwrap_or(20)
+    }
+
+    /// See [Self::idle_timeout_seconds]
+    pub fn set_idle_timeout_seconds(&mut self, idle_timeout_seconds: Option<u32>) {
+        self.idle_timeout_seconds = idle_timeout_seconds;
+    }
+
+    /// Sets the number of seconds before timing out a gRPC query
+    ///
+    /// Defaults to 5 seconds
+    pub fn query_timeout_seconds(&self) -> u32 {
+        self.query_timeout_seconds.unwrap_or(5)
+    }
+
+    /// See [Self::query_timeout_seconds]
+    pub fn set_query_timeout_seconds(&mut self, query_timeout_seconds: Option<u32>) {
+        self.query_timeout_seconds = query_timeout_seconds;
+    }
+
+    /// Number of attempts to make at a query before giving up.
+    ///
+    /// Only retries if there is a tonic-level error.
+    ///
+    /// Defaults to 3
+    pub fn query_retries(&self) -> u32 {
+        self.query_retries.unwrap_or(3)
+    }
+
+    /// See [Self::query_retries]
+    pub fn set_query_retries(&mut self, query_retries: Option<u32>) {
+        self.query_retries = query_retries;
+    }
+
+    /// Should we automatically retry transactions with corrected sequence numbers?
+    ///
+    /// Default: false
+    pub fn autofix_sequence_mismatch(&self) -> bool {
+        self.autofix_sequence_mismatch.unwrap_or(false)
+    }
+
+    /// See [Self::autofix_sequence_mismatch]
+    pub fn set_autofix_sequence_mismatch(&mut self, autofix_sequence_mismatch: Option<bool>) {
```

Comment:
Do you think this API would be more helpful with the `Option` type ?

Reply:
You mean more helpful without the `Option`? Maybe. I left the Option on as a way to say "go back to the default." It may not make as much sense in this case. I'm also considering removing this feature entirely, I think it may overall cause more harm than good.

Reply:
Yes. Sorry, these days I'm totally negating my statements.